python/pyspark/sql/catalog.py

```diff
@@ -227,15 +227,15 @@ def dropGlobalTempView(self, viewName):
     @ignore_unicode_prefix
     @since(2.0)
     def registerFunction(self, name, f, returnType=StringType()):
-        """Registers a python function (including lambda function) as a UDF
+        """Registers a Python function (including lambda function) or a wrapped/native UDF
```

Comment:
I'm really confusing when reading this document, it would be much more clear to me if we can just say
```
Registers a Python function (including lambda function) or a :class:`UserDefinedFunction`
```
This wrapping logic was added in https://github.com/apache/spark/pull/16534 , is it really worth?

Reply:
It indeed added some complexity. However, I believe nothing is blocked by #16534 now if I understand correctly.

The changes #16534 is quite nice because IMHO Python guys probably use `help()` and `dir()` more frequently then reading the API doc in the website. For the set of UDFs are provided as a library, I think that's quite worth to keep.

How about leaving this wrapper logic as is for now and then we bring this discussion back when actually something is blocked (or being too complicated) by this?

Reply:
Another idea just in case it helps:

```
Registers a Python function as a UDF or a user defined function.
```

Reply:
BTW, to be honest, I remember I gave several quick tries to get rid of the wrapper but keep the docstring correctly at that time but I failed to make a good alternative.

Might be good to try if there is a claver way to get rid of the wrapper but keep the doc.

Reply:
SGTM